File: Java-Source/Day14.java
//DAY 14 OF HACKERRANK 30 DAY CHALLENGE: SCOPE (JAVA)

import java.util.*;

//THIS IS THE CLASS USED IN THIS EXERCISE.
class Difference {
    private int[] elements;
    public int maximumDifference;

    //THESE PARTS OF THE DIFFERENCE CLASS WERE DONE FOR THE EXERCISE.
    public Difference(int[] input) {
        elements = input;
    }

    public void computeDifference() {
        int min = Integer.MAX_VALUE;
        int max = Integer.MIN_VALUE;
        for (int x = 0; x < elements.length; x++) {
            if (elements[x] > max) {
                max = elements[x];
            }
            if (elements[x] < min) {
                min = elements[x];
            }
        }
        maximumDifference = max - min;
    }
}

//THIS CLASS WAS GIVEN AND CONTAINS THE MAIN METHOD FOR WORKING WITH THE DIFFERENCE CLASS.
public class Day14 {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }
        sc.close();

        Difference difference = new Difference(a);

        difference.computeDifference();

        System.out.print(difference.maximumDifference);
    }
}

File: CPP-Source/Day14.cpp
//DAY 14 OF HACKERRANK 30 DAY CHALLENGE: SCOPE (C++)

#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

class Difference {
private:
	vector<int> elements;

public:
	int maximumDifference;

//THIS PART OF THE DIFFERENCE CLASS WAS FILLED OUT FOR THE EXERCISE.
	Difference(vector<int> input) {
		elements = input;
	}

	void computeDifference() {
		int min = 101;//Higher than range given of 1<=x<=100. Forces any input to change its value.
		int max = 0;//Lower than range given of 1<=x<=100 Forces any input to change its value.

		for (int x = 0; x < elements.size(); x++) {
			if (elements[x] < min) {
				min = elements[x];
			}
			if (elements[x] > max) {
				max = elements[x];
			}
		}

		maximumDifference = max - min;
	}
}; 

//MAIN METHOD GIVEN USED FOR TESTING THE CLASS THAT WAS COMPLETED.
int main() {
	int N;
	cin >> N;

	vector<int> a;

	for (int i = 0; i < N; i++) {
		int e;
		cin >> e;

		a.push_back(e);
	}

	Difference d(a);

	d.computeDifference();

	cout << d.maximumDifference;

	return 0;
}